### PR TITLE
CB-8734 add support for firefoxos

### PR DESCRIPTION
**Pull request by marti1125**

### CODE REVIEW

File: buildbot-conf/cordova.conf [hunk 1]
```diff
@@ -300,10 +300,13 @@ c['builders'].extend([
...
     BuilderConfig(name='cordova-blackberry-osx', slavenames=OSX_SLAVES,     factory=cordova_plugins_all,     properties={'platform': 'blackberry10', 'slaveos': OSX}),
+    BuilderConfig(name='cordova-firefoxos-osx',  slavenames=OSX_SLAVES,     factory=cordova_plugins_all,     properties={'platform': 'firefoxos',    'slaveos': OSX}),
     BuilderConfig(name='cordova-windows',        slavenames=WINDOWS_SLAVES, factory=cordova_plugins_windows, properties={'platform': 'windows',      'slaveos': WINDOWS}),
     BuilderConfig(name='cordova-wp8',            slavenames=WINDOWS_SLAVES, factory=cordova_plugins_all,     properties={'platform': 'wp8',          'slaveos': WINDOWS}),
     BuilderConfig(name='cordova-android-win',    slavenames=WINDOWS_SLAVES, factory=cordova_plugins_all,     properties={'platform': 'android',      'slaveos': WINDOWS}),
     BuilderConfig(name='cordova-blackberry-win', slavenames=WINDOWS_SLAVES, factory=cordova_plugins_all,     properties={'platform': 'blackberry10', 'slaveos': WINDOWS}),
+    BuilderConfig(name='cordova-firefoxos-win',  slavenames=WINDOWS_SLAVES, factory=cordova_plugins_all,     properties={'platform': 'firefoxos',    'slaveos': WINDOWS}),
```
Comment (dblotsky) [inline buildbot-conf/cordova.conf]: Please add the new builders to the scheduler on [line 315](https://github.com/apache/cordova-medic/blob/master/buildbot-conf/cordova.conf#L315).

File: src/build/makers/firefoxos.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+var shell        = require('shelljs'),
+    path         = require('path'),
+    error_writer = require('./error_writer'),
+    n            = require('ncallbacks'),
+    fs           = require('fs'),
+    mspec        = require('./mobile_spec');
```
Comment (dblotsky) [inline src/build/makers/firefoxos.js]: This guy is a directory, so the script fails with `Error: Cannot find module './mobile_spec'`.

**Comment by dblotsky** [comment]
Please make sure that the build scripts run on their own.

**Comment by marti1125** [comment]
ok :D sorry for inconvenients

**Comment by marti1125** [comment]
Is it correct?
bs

File: buildbot-conf/cordova.conf [hunk 3]
```diff
@@ -327,6 +330,7 @@ c['schedulers'].extend([
             'cordova-wp8',
             'cordova-android-win',
             'cordova-blackberry-win',
+            'cordova-firefoxos',
```
Comment (dblotsky) [inline buildbot-conf/cordova.conf]: This is a list of builders that you define above, so you will want `cordova-firefoxos-win` and `cordova-firefoxos-osx` here.

**Comment by dblotsky** [comment]
Nope. Sorry, I didn't clarify. It's great that you have a medic installation running locally! However what I mean is this: you can run all the scripts in medic on your own from the command line like so:

    $ node cordova-medic/build_firefoxos.js

Running this script manually will help weed out any bugs in the code without needing to run a full setup of medic. Since medic just runs this script the same way, you can be confident that if running it manually is good, then it will also be good in medic.

**Comment by marti1125** [comment]
after run: $ npm install

λ node build_firefoxos.js

module.js:340
    throw err;
          ^
Error: Cannot find module '../../config'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (G:\cordova-medic\src\couchdb\interface.js:19:14)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)

G:\cordova-medic (CB-8734)
λ node build_android.js

module.js:340
    throw err;
          ^
Error: Cannot find module '../../config'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (G:\cordova-medic\src\couchdb\interface.js:19:14)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)

**Comment by dblotsky** [comment]
Yep, that is expected. Unfortunately the new medic code still has some baggage from the old design that has not yet been refactored. Specifically, there is a [config download step](https://github.com/apache/cordova-medic/blob/master/buildbot-conf/cordova.conf#L238), where medic's config is downloaded from a master to a slave. You will need to manually create this file based on the [sample file](https://github.com/apache/cordova-medic/blob/master/buildbot-conf/cordova-config.json.sample), since the build scripts all depend on it. The same goes for the [repos file](https://github.com/apache/cordova-medic/blob/master/buildbot-conf/cordova-repos.json).

**Comment by marti1125** [comment]
the unique error is about path of mobilespec

G:\cordova-medic (CB-8734)
λ node build_firefoxos.js
G:\cordova-medic\.git G:\cordova-medic
G:\cordova-medic
G:\cordova-medic\bin G:\cordova-medic
G:\cordova-medic
G:\cordova-medic\buildbot-conf G:\cordova-medic
G:\cordova-medic
G:\cordova-medic\node_modules G:\cordova-medic
G:\cordova-medic
G:\cordova-medic\runner G:\cordova-medic
G:\cordova-medic
G:\cordova-medic\src G:\cordova-medic
G:\cordova-medic
[FIREFOXOS ERROR]: (sha: Firefox)
Exception thrown modifying FirefoxOS mobile spec application.
create must have failed as output path does not exist.path  G:\cordova-medic\mobilespec\platforms\firefoxos
FirefoxOS test prepare failed

**Comment by dblotsky** [comment]
You need to follow the same steps that Medic follows to use the build scripts. That is:

```bash
    $ # make sure all the relevant repositories are cloned, npm-installed, and npm-linked
    $ node cordova-mobile-spec/createmobilespec/createmobilespec.js --firefoxos --copywww --skiplink mobilespec
    $ node cordova-medic/updateconfig.js --firefoxos
    $ node cordova-medic/build_firefoxos.js
```

If it's still failing for you, please post your directory layout and outputs on the mailing list and we can debug there.

**Comment by marti1125** [comment]
Hi, in firefoxos I have this  //fs.writeFileSync(path.join(output, 'assets', 'www','autotest','pages', 'medic.json'),medic_config,'utf-8'); this file doesn't exist

**Comment by marti1125** [comment]
I had to move mobilespec to inside of cordova-medic
λ node build_firefoxos.js
G:\cordova-medic\.git G:\cordova-medic
G:\cordova-medic
G:\cordova-medic\bin G:\cordova-medic
G:\cordova-medic
G:\cordova-medic\buildbot-conf G:\cordova-medic
G:\cordova-medic
G:\cordova-medic\mobilespec G:\cordova-medic
G:\cordova-medic
G:\cordova-medic\node_modules G:\cordova-medic
G:\cordova-medic
G:\cordova-medic\runner G:\cordova-medic
G:\cordova-medic
G:\cordova-medic\src G:\cordova-medic
G:\cordova-medic
[FIREFOXOS] Modifying Cordova firefoxos application. (sha: FirefoxOS__master__1427406728)

**Comment by dblotsky** [comment]
You need to call `createMedicJson`, defined [here](https://github.com/apache/cordova-medic/blob/master/src/utils/createMedicJson.js#L4), similarly to how the other build scripts call it (example [here](https://github.com/apache/cordova-medic/blob/master/build_ios.js)).

**Pull request closed without merging**